deltametrics/io.py

```diff
@@ -0,0 +1,217 @@
+        """Should read data into memory.
+        """
+        return
+
+    @abc.abstractmethod
+    def write(self):
+        """Should write the data to file.
+
+        Take a :obj:`~deltametrics.cube.Cube` and write it to file.
+        """
+        return
+
+    @abc.abstractmethod
+    def __getitem__(self):
+        """Should slice the data from file.
+        """
+        return
+
+    @property
+    @abc.abstractmethod
+    def keys(self):
+        """Should link to all key _names_ stored in file.
+        """
+        return
+
+
+class NetCDFIO(BaseIO):
+    """Utility for consistent IO with netCDF files.
+
+    This module wraps calls to the netCDF4 python module in a consistent API,
+    so the user can work seamlessly with either netCDF4 files or HDF5 files.
+    The public methods of this class are consistent with
+    :obj:`~deltametrics.utils.HDFIO`.
+    """
+
+    def __init__(self, data_path, write=False):
+        """Initialize the NetCDFIO handler.
+
+        Initialize a connection to a NetCDF file.
+
+        Parameters
+        ----------
+        data_path : `str`
+            Path to file to read or write to.
+
+        write : `bool`, optional
+            Whether to allow writing to an existing file. Set to False by
+            default, if a file already exists at ``data_path``, writing is
+            disabled, unless ``write`` is set to True.
+        """
+
+        super().__init__(data_path=data_path, write=write)
+
+        self.type = 'netcdf'
+        self._in_memory_data = {}
+
+    def connect(self):
+        """Connect to the data file.
+
+        Initialize the file if it does not exist, or simply ``return`` if the
+        file already exists.
+
+        .. note::
+            This function is automatically called during initialization of any
+            IO object, so it is not necessary to call it directly.
+
+        """
+        if not os.path.isfile(self.data_path):
+            self.dataset = netCDF4.Dataset(
+                self.data_path, "w", format="NETCDF4")
+        else:
+            if self.write:
+                self.dataset = netCDF4.Dataset(self.data_path, "r+")
+            else:
+                self.dataset = netCDF4.Dataset(self.data_path, "r")
+
+    def read(self, var):
+        """Read variable from file and into memory.
+
+        Converts `variables` in netCDF file to `ndarray` for coersion into a
+        :obj:`~deltametrics.cube.Cube` instance.
+
+        Parameters
+        ----------
+        var : `str`
+            Which variable to load from the file.
+        """
+        try:
+            _arr = self.dataset.variables[var]
+        except ValueError as e:
+            raise e
+        self._in_memory_data[var] = np.array(_arr, copy=True)
+
+    def write(self):
+        """Write data to file.
+
+        Take a :obj:`~deltametrics.cube.Cube` and write it to file.
+
+        .. warning::
+            Not Implemented.
+
+        """
+        raise NotImplementedError
```

Comment:
Woof. This is going to be a hard one to do, just because there's all kinds of contingencies to  plan for, including how people want to compress and write to disk.

Reply:
hahah that's fine, I'm not worried about leaving it not implemented for now. We can worry about it in the future. 

Ultimately, I think we should *think* about making the package flexible for lots of use cases, but primarily we want to make the parts that we need first, and worry about other stuff later.

Reply:
I agree, let's just leave this unimplemented for now.